src/controllers/user/Controller.ts

```diff
@@ -2,16 +2,19 @@ import { Request, Response, NextFunction } from 'express';
 import * as jwt from 'jsonwebtoken';
 import configuration from '../../config/configuration';
 import UserRepository from '../../repositories/user/UserRepository';
+import { hashPassword } from '../../utils/helpers';
 import * as bcrypt from 'bcrypt';
 
 const userRepository: UserRepository = new UserRepository();
 class User {
-    // FUNCTION TO GET ALL USER
+    // FUNCTION TO GET ALL USERS
     getAll = async (req: Request, res: Response, next: NextFunction) => {
         try {
             console.log('Get request...!');
-            const userData = await userRepository.findDoc({});
-            res.status(200).send(userData);
+            const userData = await userRepository.findDoc(req.query);
+            const documents = await userRepository.countDoc();
```

Comment:
use Promise.All for independent db calls for better performance
for        const userData = await userRepository.findDoc(req.query);
            const documents = await userRepository.countDoc();

Reply:
Done with all the changes requested.